src/fu-udev-backend.c

```diff
@@ -41,7 +41,8 @@ fu_udev_backend_device_remove(FuUdevBackend *self, GUdevDevice *udev_device)
 	/* find the device we enumerated */
 	device_tmp =
 	    fu_backend_lookup_by_id(FU_BACKEND(self), g_udev_device_get_sysfs_path(udev_device));
-	if (device_tmp != NULL) {
+	if (device_tmp != NULL &&
+	    !fu_device_has_flag(device_tmp, FWUPD_DEVICE_INTERNAL_FLAG_NO_AUTO_REMOVE)) {
```

Comment:
i think you want the same thing in fu-**usb**-backend.c too.

Reply:
Sure added for  fu-usb-backend.c too  , what are the the benefits to add it to usb backend ?
for later platforms we want to extend this to multiport where both ports may or may not share same nvm payload, is there a place to keep a check on the extracted nvm files not to be cleaned up until all the ports nvm is updated ? currently we dont have this issue but just checking to extend the functionality for future...

Reply:
> what are the the benefits to add it to usb backend

To make the codebase consistent. _NO_AUTO_REMOVE is the kind of thing I could see being used for USB devices too.

> , is there a place to keep a check on the extracted nvm files not to be cleaned up until all the ports nvm is updated

Back a bit; how do you mean? Surely each "port" would have a unique FuDevice and associated GUID?